---
bool fetch_ret    # fetching result
float32[] action  # single action
uint8 queue_size  # the size of the queue after fetching